src/editor/lib/blocks/ucd-theme-career/edit.js

```diff
@@ -0,0 +1,197 @@
+import { html } from '../../utils';
+import { useBlockProps } from '@wordpress/block-editor';
+import {
+  Button,
+  TextControl,
+  Modal,
+  SelectControl,
+  DatePicker
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
+
+export default ( props ) => {
+  const { attributes, setAttributes } = props;
+  const blockProps = useBlockProps();
+
+  // modal state
+  const startingModalData = {
+    title: attributes.title || '', 
+    link: attributes.link || '', 
+    salaryMin: attributes.salaryMin || '', 
+    salaryMax: attributes.salaryMax || '', 
+    employmentType: attributes.employmentType || 'FULL_TIME',
+    salaryFrequency: attributes.salaryFrequency || 'HOUR',
+    finalFilingDate: attributes.finalFilingDate || '',
+  };
+  const [ modalIsOpen, setModalOpen ] = useState( false );
+  const [ modalMode, setModalMode ] = useState( 'Add' );
+  const [ modalData, setModalData ] = useState( startingModalData );
+
+  // check if all data is empty (excluding default values)
+  const isEmpty = Object.values(startingModalData).every(x => x === null || x === '' || x === 'FULL_TIME' || x === 'HOUR');
+  setAttributes({isEmpty});
+
+  // modal validation
+  const modalCanSave = (() => {
+    if ( 
+      !modalData || 
+      !modalData.title || 
+      !modalData.link || 
+      !modalData.salaryMin ||
+      !modalData.salaryMax ||
+      !modalData.salaryFrequency ||
+      !modalData.finalFilingDate
+      ) {
+        return false
+      }
+    return true;
+  })();
+
+  // dropdown options
+  const salaryFreqencies = [
+    {label: 'Hour', value: 'HOUR'},
+    {label: 'Day', value: 'DAY'},
+    {label: 'Week', value: 'WEEK'},
+    {label: 'Month', value: 'MONTH'},
+    {label: 'Year', value: 'YEAR'}
+  ];
+
+  const employmentTypes = [
+    {label: 'Full-Time', value: 'FULL_TIME'},
+    {label: 'Part-Time', value: 'PART_TIME'},
+    {label: 'Contract', value: 'CONTRACT'},
+    {label: 'Temporary', value: 'TEMPORARY'},
+    {label: 'Internship', value: 'INTERNSHIP'},
+  ];
+
+  // currency formatter
+  const currency = new Intl.NumberFormat('en-US', {
+    style: 'currency',
+    currency: 'USD'
+  });
+
+  const onCareerClicked = (e) => {
+    setModalMode('Edit');
+    setModalOpen(true);
+  }
+
+  // modal data change events
+  const onModalTitleChange = (title) => {
+    const data = {...modalData, title};
+    setModalData(data);
+    setAttributes({title});
```

Comment:
You don't need to use `setAttributes` AND `setModalData` on your `onChange` listeners. React likes to divide a line between [state (setModalData) and props (setAttributes)](https://reactjs.org/docs/faq-state.html). in my experience, react developers have FEELINGS about this.

So in this case, I would remove all of the `setAttributes` functions on your event listeners, and then move them to `onModalSave`  and combine them to be just `setAttributes(modalData)`